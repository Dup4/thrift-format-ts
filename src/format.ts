/* eslint-disable @typescript-eslint/no-unused-vars */
import { IOptions } from "./types";

class StructLine {
  number = "";
  paramOptions = "";
  paramName = "";
  defaultValue = "";
  description = "";
  annotation = "";
}

class EnumLine {
  number = "";
  paramName = "";
  hasDescription = "";
  description = "";
  annotation = "";
}

export function formatThrift(
  content: string,
  options?: IOptions,
): [string, boolean] {
  const defaultOptions: IOptions = {};
  content = deleteExtraEmptyLines(content, options || defaultOptions);
  return [content, true];
}

function deleteExtraEmptyLines(content: string, options?: IOptions) {
  const c = content.split("\n");
  const res: string[] = [];

  for (const l of c) {
    if (l !== "") {
      res.push(l);
    } else if (res.length === 0 || res.at(-1) !== "") {
      res.push(l);
    }
  }

  return res.join("\n");
}
