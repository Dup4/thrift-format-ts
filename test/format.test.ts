import { describe, expect, it } from "vitest";
import { formatThrift } from "@/format";

import * as fs from "fs";
import path from "path";

function getFullPath(file_name: string): string {
  return path.resolve(__dirname, "test_data", file_name).toString();
}

describe("should", () => {
  it("exported", () => {
    const thriftContent = fs.readFileSync(getFullPath("a.thrift")).toString();
    expect(
      formatThrift(thriftContent, { enableLabFormat: true }),
    ).matchSnapshot();
  });

  it("exported", () => {
    const thriftContent = fs
      .readFileSync(getFullPath("ThriftTest.thrift"))
      .toString();
    expect(
      formatThrift(thriftContent, { enableLabFormat: true }),
    ).matchSnapshot();
  });
});
