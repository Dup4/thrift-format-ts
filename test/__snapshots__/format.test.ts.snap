// Vitest Snapshot v1

exports[`should > exported 1`] = `
"struct A {
    1: required string a,
}

struct B {
    1: required string b,
}"
`;
